identity/src/androidTest/java/com/android/identity/IdentityCredentialTest.java

```diff
@@ -0,0 +1,172 @@
+          throws IdentityCredentialException {
+    final int usesPerKey = 3;
+    ProvisioningTest.createCredential(store, CREDENTIAL_NAME);
+    try {
+      IdentityCredential credential = Preconditions.checkNotNull(getCredentialIfPresent());
+
+      storeFakeStaticAuthData(credential, usesPerKey, numAuthKeys, staticAuthDataSizeBytes);
+
+      // Check that this doesn't throw (it did throw in internal b/234563696).
+      try {
+        getCredentialIfPresent();
+      } catch (RuntimeException e) {
+        String msg = String.format(Locale.US,
+                "Failed to load credential for %d authKeys with %d bytes static auth data each",
+                numAuthKeys, staticAuthDataSizeBytes);
+        throw new IdentityCredentialException(msg, e);
+      }
+    } finally {
+      deleteCredentialFromStoreIfPresent();
+    }
+  }
+
+  @Before
+  public void setUp() {
+    deleteCredentialFromStoreIfPresent(); // clean up in case a previous test didn't
+  }
+
+  @After
+  public void tearDown() {
+    deleteCredentialFromStoreIfPresent();
+    store = null;
+  }
+
+  private static void storeFakeStaticAuthData(IdentityCredential credential,
+          int maxUsesPerKey, int numKeys, int staticAuthDataSizeBytes) {
+    Random random = new Random(31337 + numKeys);
+    credential.setAvailableAuthenticationKeys(numKeys, maxUsesPerKey);
+    Collection<X509Certificate> authKeys = credential.getAuthKeysNeedingCertification();
+    assertEquals(numKeys, authKeys.size());
+
+    for (X509Certificate authKey : authKeys) {
+      byte[] staticAuthData = new byte[staticAuthDataSizeBytes];
+      random.nextBytes(staticAuthData);
+      Calendar expirationDate = Calendar.getInstance();
+      expirationDate.add(Calendar.YEAR, 1);
+      try {
+        credential.storeStaticAuthenticationData(authKey, expirationDate, staticAuthData);
+      } catch (UnknownAuthenticationKeyException e) {
+        throw new AssertionFailedError("Auth authKey not recognized: " + e);
+      }
+    }
+  }
+
+  private @Nullable IdentityCredential getCredentialIfPresent() {
+    try {
+      return store.getCredentialByName(CREDENTIAL_NAME, CIPHER_SUITE);
+    } catch (CipherSuiteNotSupportedException e) {
+      throw new AssertionFailedError("Unexpected exception: " + e);
+    }
+  }
+
+  private void deleteCredentialFromStoreIfPresent() {
+    store.deleteCredentialByName(CREDENTIAL_NAME);
+    // The above method is deprecated, but works even if the credential is corrupted, whereas the
+    // following can throw RuntimeException in that case (e.g. b/234563696):
+    // IdentityCredential credential = getCredentialIfPresent();
+    // if (credential != null) {
+    //   credential.delete("fake challenge for deletion".getBytes(UTF_8));
+    // }
+  }
+
+  // parameters: { store, storeName }
+  @Parameters(name = "{1}")
+  public static Collection<Object[]> stores() {
+    List<IdentityCredentialStore> stores = new ArrayList<>();
```

Comment:
This is shadowing the method name. I'm not up-to-speed on typical Java style, but shadowing generally seems like a bad idea to me.

Reply:
Personally I see no issue with a variable having the same name as a method, since variable name references are radically different from method invocations. But, stores() renamed:

stores() -> parameters()
store -> resultStores